src/core/main.c

```diff
@@ -1523,7 +1523,7 @@ int main(int argc, char *argv[]) {
                 /* Try to figure out if we can use colors with the console. No
                  * need to do that for user instances since they never log
                  * into the console. */
-                log_show_color(colors_enabled());
+                log_show_color(colors_enabled_no_tty(true));
                 make_null_stdio();
```

Comment:
This code path is also taken in containers, and systemd should honour $SYSTEMD_COLORS there. Also, we do want to bother with caching, because colors_enabled() is called from many other places. So we want to prime the cache with a proper value. I don't see how the code could be reworked with getenv_term_is_dump() without repeating much of the logic in both call stacks.